src/components/header/header.tsx

```diff
@@ -19,10 +19,14 @@ export const Header = () => {
         [walletAddress]
     );
     const [isDropdownOpen, setIsDropdownOpen] = useState(false);
+    console.log(isDropdownOpen);
 
     const handleDropdownClick = () => setIsDropdownOpen(value => !value);
     const handleConnectClick = () => connectModal.open();
-    const handleDisconnectClick = () => tonConnectUI.disconnect();
+    const handleDisconnectClick = () => {
+        tonConnectUI.disconnect();
+        setIsDropdownOpen(false);
```

Comment:
it's a bad fix. You don't need to handle dropdown click if the button is not a dropdown.